File: app/models/risk_assessment.py
from datetime import datetime, timezone
from enum import Enum

from pydantic import BaseModel, Field


class RiskLevel(str, Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"


class RiskAssessment(BaseModel):
    session_id: str
    agent_id: str
    risk_score: int
    risk_level: RiskLevel
    finding_count: int
    assessed_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc)
    )

File: frontend/src/types/riskAssessment.ts
/**
 * riskAssessment.ts — Dynamic Risk Assessment domain types.
 *
 * Type contracts for process-local risk assessments produced by the
 * Dynamic Risk Engine (ADR-018).
 *
 * Backend sources:
 *   GET /api/v1/risk-assessments
 *   GET /api/v1/risk-assessments/:session_id
 *
 * Feature gate: RISK_ENGINE_ENABLED (src/config/featureFlags.ts)
 */

export type RiskLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'

export interface RiskAssessment {
  session_id: string
  agent_id: string
  risk_score: number
  risk_level: RiskLevel
  finding_count: number
  assessed_at: string
}
